.azure-pipelines/ultimate-pipeline.yml

```diff
@@ -898,6 +945,144 @@ stages:
       displayName: Check logs for errors
 
+- stage: exploration_tests_linux
+  condition: and(succeeded(), eq(variables['isScheduledBuild'], 'False'), or(eq(stageDependencies.check_changes.outputs['check_changes_job.check_changes_step.IsProfilerChanged'],'True'), eq(stageDependencies.check_changes.outputs['check_changes_job.check_changes_step.IsDebuggerChanged'], 'True')))
+  dependsOn: [build_linux, check_changes]
+  jobs:
+  - template: steps/update-github-status-jobs.yml
+    parameters:
+      jobs: [Test]
+
+  - job: generator
+    variables:
+      ISDEBUGGERCHANGED: $[stageDependencies.check_changes.check_changes_job.outputs['check_changes_step.IsDebuggerChanged']]
+      ISPROFILERCHANGED: $[stageDependencies.check_changes.check_changes_job.outputs['check_changes_step.IsProfilerChanged']]
+    pool:
+      vmImage: ubuntu-18.04
+    steps:
+      - bash: |
+          matrix="{"
+
+          declare -a explorationTestUseCases=()
+          if [ $ISDEBUGGERCHANGED == "True" ]; then 
+            explorationTestUseCases+=("debugger")
+          fi
+
+          if [ $ISPROFILERCHANGED == "True" ]; then 
+            explorationTestUseCases+=("continuousProfiler")
+          fi
+
+          add_to_matrix () {
+            name="${baseImage//./_}_${publishTargetFramework//./_}_${explorationTestUseCase//./_}_${explorationTestName}"
+            matrix="${matrix} '${name}':"
+            matrix="${matrix} { 'baseImage': '${baseImage}', "
+            matrix="${matrix} 'publishTargetFramework': '${publishTargetFramework}', "
+            matrix="${matrix} 'explorationTestUseCase': '${explorationTestUseCase}', "
+            matrix="${matrix} 'explorationTestName': '${explorationTestName}' },"
+          }
+
+          for baseImage in "debian" "alpine" ; do
+            for explorationTestUseCase in ${explorationTestUseCases[@]} ; do
+
+              publishTargetFramework="netcoreapp2.1"
+              for explorationTestName in "eshoponweb" "protobuf" ; do
+                add_to_matrix
+              done
+
+              publishTargetFramework="netcoreapp3.1"
+              explorationTestName="cake"
+              add_to_matrix
+
+              publishTargetFramework="net5.0"
+              explorationTestName="cake"
+              add_to_matrix
+
+              publishTargetFramework="net6.0"
+              for explorationTestName in  "cake" "swashbuckle"; do
+                add_to_matrix
+              done
+            done
+          done
+          matrix="${matrix::-1} }"
+          echo "##vso[task.setVariable variable=values;isOutput=true]${matrix}"
+        name: mtrx
+
+
+  - job: Test
+    dependsOn: generator
+    strategy:
+      matrix: $[ dependencies.generator.outputs['mtrx.values'] ]
+
+    variables:
+      TestAllPackageVersions: true
+      IncludeMinorPackageVersions:  $[or(eq(variables.isMainOrReleaseBranch, 'true'), eq(variables.perform_comprehensive_testing, 'true'))]
+
```

Comment:
Seems to be copy/pasted over from Tracer tests, but I don't think we need these here?

```suggestion
```